pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -210,6 +242,26 @@ func (cache *claimInfoCache) hasPodReference(UID types.UID) bool {
 	return false
 }
 
+func (cache *claimInfoCache) addPodReference(claimName, namespace string, podUID types.UID) {
+	cache.Lock()
+	defer cache.Unlock()
+
+	claimInfo, ok := cache.claimInfo[claimName+namespace]
+	if ok {
+		claimInfo.PodUIDs.Insert(string(podUID))
+	}
+}
+
+func (cache *claimInfoCache) deletePodReference(claimName, namespace string, podUID types.UID) {
+	cache.Lock()
+	defer cache.Unlock()
+
+	claimInfo, ok := cache.claimInfo[claimName+namespace]
+	if ok {
+		claimInfo.PodUIDs.Delete(string(podUID))
+	}
+}
+
```

Comment:
Should these return errors if not in the cache?

Reply:
To be clear -- I'm not saying they have to, it just feels inconsistent for some of these calls to return an error and others to silently do nothing.

Reply:
I think it makes sense. Thanks for catching it. With this check the API will look more consistend.

Reply:
The alternative is that we don’t lock anywhere inside the claimInfoCahe abstraction itself and instead call m.cache.Lock() / m.cache.Unlock() where appropriate when being used.

Reply:
Sorry, this comment but dropped in the wrong place, but you get the point.

Reply:
> The alternative is that we don’t lock anywhere inside the claimInfoCahe abstraction itself and instead call m.cache.Lock() / m.cache.Unlock() where appropriate when being used.

I thought about it and even tried it in another branch yesterday: https://github.com/bart0sh/kubernetes/commit/8c66f3a9cb7b251fdf9e52809689a93a9868789f

I have mixed fillings about it. It helps to increase number of locks for sure, but it's easier to forget to unlock the cache as unlocking can't be deferred.

Reply:
I'm not saying we should do this, but you can use this trick to insert defers at arbitrary points in the code:

Instead of:
```
someCode1()
someCode2()

Lock()
if err := whatever() {
   Unlock()
   return err
}
Unlock()
return nil
```
you can do:
```
someCode1()
someCode2()

func() {
  Lock()
  defer Unlock()
  
  if err := whatever() {
   return err
  }
  return nil
}()
```

Reply:
Thanks for the tip. Yes, I can do that, but readability would safer. Your first example is much more readable in my opinion. Do you think it makes sense to do?